apm-agent-core/src/main/java/co/elastic/apm/agent/report/AbstractIntakeApiHandler.java

```diff
@@ -103,18 +106,44 @@ protected boolean shouldEndRequest() {
     protected HttpURLConnection startRequest(String endpoint) throws IOException {
         final HttpURLConnection connection = apmServerClient.startRequest(endpoint);
         if (connection != null) {
-            if (logger.isDebugEnabled()) {
-                logger.debug("Starting new request to {}", connection.getURL());
+            try {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Starting new request to {}", connection.getURL());
+                }
+                connection.setRequestMethod("POST");
+                connection.setDoOutput(true);
+                connection.setChunkedStreamingMode(DslJsonSerializer.BUFFER_SIZE);
+                connection.setRequestProperty("Content-Encoding", "deflate");
+                connection.setRequestProperty("Content-Type", "application/x-ndjson");
+                connection.setUseCaches(false);
+                connection.connect();
+                os = new DeflaterOutputStream(connection.getOutputStream(), deflater);
+                os.write(metaData);
+            } catch (IOException e) {
+                logger.error("Error trying to connect to APM Server. If SSL is enabled, set logging level to info " +
+                    "for detailed SSL configurations available for the current connection.");
+                if (logger.isInfoEnabled() && connection instanceof HttpsURLConnection) {
+                    HttpsURLConnection httpsURLConnection = (HttpsURLConnection) connection;
+                    try {
+                        logger.info("Cipher suite used for this connection: {}", httpsURLConnection.getCipherSuite());
+                    } catch (Exception e1) {
+                        SSLSocketFactory sslSocketFactory = httpsURLConnection.getSSLSocketFactory();
```

Comment:
Because it is not interesting for our purpose, it's just a premature check on a non-initialized connection done for info logging. I am falling back to doing what I need to do on non-initialized connections.